### PR TITLE
updates for managed cluster labels

### PR DESCRIPTION
**Pull request by dockerymick**
https://github.com/stolostron/backlog/issues/27016

### CODE REVIEW

File: observability/design_grafana.adoc [hunk 1]
```diff
@@ -102,6 +106,110 @@ annotations:
...
+      - installer.name
+      - installer.namespace
+      - local-cluster
+      - name
+    labels:
+      - cloud
+      - vendor
+----
+
+The labels that are enabled are displayed in the drop-down filter on the _ACM - Clusters Overview_ Grafana dashboard. The values are from the `managed_cluster_labels` metric, depending on the `label` key value that is selected.
```
Comment (dislbenn) [inline observability/design_grafana.adoc]: The metric name is prefixed with `acm`.

```suggestion
The labels that are enabled are displayed in the drop-down filter on the _ACM - Clusters Overview_ Grafana dashboard. The values are from the `acm_managed_cluster_labels` metric, depending on the `label` key value that is selected.
```

File: observability/design_grafana.adoc [hunk 2]
```diff
@@ -102,6 +106,110 @@ annotations:
 +
 After you complete your updates for the ConfigMap, you can install it to import the dashboard to the Grafana instance.
 
+[#using-managed-cluster-labels]
+== Using managed cluster labels in Grafana
+
+When Observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is createed in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels that are maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, Observability ignores out-of-the-box labels in the `observability-managed-cluster-label-allowlist` ConfigMap.
```
Comment (dislbenn) [inline observability/design_grafana.adoc]: Created was misspelled.

```suggestion
When Observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, Observability ignores out-of-the-box labels in the `observability-managed-cluster-label-allowlist` ConfigMap.
```

File: observability/design_grafana.adoc [hunk 3]
```diff
@@ -102,6 +106,110 @@ annotations:
...
+//is there a command to run for this verification? 
+From the Grafana dashboard, verify that the label is listed as a valiue in the _Label_ drop-down menu.
+
+
+[#disabling-managed-cluster-labels]
+=== Disabling managed cluster labels
+
+
+
+
```
Comment (dislbenn) [inline observability/design_grafana.adoc]: This section is empty for disabling the managed cluster labels.

Comment (dockerymick) [reply observability/design_grafana.adoc]: My bad! I thought i added it. Adding it soon

File: observability/design_grafana.adoc [hunk 4]
```diff
@@ -102,6 +106,110 @@ annotations:
...
+----
+
+The ConfigMap resyncs after 30 seconds to ensure that the cluster labels are updated. After you update the ConfigMap, check the `observability-rbac-query-proxy` pod logs in the `open-cluster-management-observability` namespace to verify where the label is listed. The following information might be displayed in the pod log:
+
+----
+enabled managedcluster labels: <label>
+----
+
+//is there a command to run for this verification? 
+From the Grafana dashboard, verify that the label is listed as a valiue in the _Label_ drop-down menu.
```
Comment (dislbenn) [inline observability/design_grafana.adoc]: Value is misspelled. Also, currently, there is no command to run for this verification. The user can only go to Grafana to view the _Label_ drop-down within the `ACM - Cluster Overview` dashboard. Generating a command could be a future enhancement that we can look into.

```suggestion
//is there a command to run for this verification? 
From the Grafana dashboard, verify that the label is listed as a value in the _Label_ drop-down menu.
```

Comment (dockerymick) [reply observability/design_grafana.adoc]: Updated

File: observability/design_grafana.adoc [hunk 5]
```diff
@@ -102,6 +106,128 @@ annotations:
...
+----
+
+The labels that are enabled are displayed in the drop-down filter on the _ACM - Clusters Overview_ Grafana dashboard. The values are from the `acm_managed_cluster_labels` metric, depending on the `label` key value that is selected.
+
+Any label that is listed in the `ignore_labels` keylist of the ConfigMap is removed from the drop-down filter on the _ACM - Clusters Overview_ Grafana dashboard.
+
+[#enabling-adding-managed-cluster-labels]
+=== Enabling and adding managed cluster labels
+
+When you add a new managed cluster label to the `observability-managed-cluster-label-allowlist` ConfigMap, you are enabling the cluster. Add a unique label to the hub cluster, or managed cluster object that is associated with the managed cluster fleet. For example, if you add the label, `department=finance` to a managed cluster, the ConfigMap is updated and might resemble the following changes:
```
Comment (dislbenn) [inline observability/design_grafana.adoc]: For this line, `"When you add a new managed cluster label to the observability-managed-cluster-label-allowlist ConfigMap, you are enabling the cluster"`, you're not actually enabling the cluster. When the label is added, that label becomes available within Grafana for the user to begin filtering from.

File: observability/design_grafana.adoc [hunk 6]
```diff
@@ -102,6 +106,128 @@ annotations:
...
+      - name
+    labels:
+      - cloud
+      - department
+      - vendor
+----
+
+Enable a managed cluster label that is already disabled by removing the label from the `ignore_labels` list in the `observability-managed-cluster-label-allowlist` ConfigMap.
+
+For example, enable the `local-cluster` and `name` labels, Your `observability-managed-cluster-label-allowlist` ConfigMap might resemble the following content:
```
Comment (dislbenn) [inline observability/design_grafana.adoc]: `Your` is capitalized, this should be lowercase.
```suggestion
For example, enable the `local-cluster` and `name` labels, your `observability-managed-cluster-label-allowlist` ConfigMap might resemble the following content:
```

File: observability/design_grafana.adoc [hunk 7]
```diff
@@ -102,6 +106,128 @@ annotations:
...
+      - cloud
+      - vendor
+ ----
+
+Check the `observability-rbac-query-proxy` pod logs in the `open-cluster-management-observability` namespace to verify where the label is listed. The following information might be displayed in the pod log:
+
+----
+disabled managedcluster label: <label>
+----
+
```
Comment (dislbenn) [inline observability/design_grafana.adoc:233]: It seems like the format might've gotten thrown off here.

Screenshot 2023-01-31 at 3 51 28 PM

File: observability/design_grafana.adoc [hunk 8]
```diff
@@ -102,6 +106,128 @@ annotations:
 +
 After you complete your updates for the ConfigMap, you can install it to import the dashboard to the Grafana instance.
 
+[#using-managed-cluster-labels]
+== Using managed cluster labels in Grafana
+
+When Observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, Observability ignores out-of-the-box labels in the `observability-managed-cluster-label-allowlist` ConfigMap.
```
Comment (oafischer) [inline observability/design_grafana.adoc]: IBM Style recommends not using out of the box as a phrase. Let's try to find a replacement.

image

File: observability/design_grafana.adoc (hunk 6 above)
Comment (oafischer) [inline observability/design_grafana.adoc]:
```suggestion
For example, enable the `local-cluster` and `name` labels. Your `observability-managed-cluster-label-allowlist` ConfigMap might resemble the following content:
```

Typo (I think)

File: observability/design_grafana.adoc (hunk 5 above)
Comment (dockerymick) [reply observability/design_grafana.adoc]: Ahh understood. Thanks for clarifying!

File: observability/design_grafana.adoc (hunk 6 above)
Comment (dockerymick) [reply observability/design_grafana.adoc]: accepting suggestion

File: observability/design_grafana.adoc (hunk 5 above)
Comment (dockerymick) [reply observability/design_grafana.adoc]: updated

File: observability/design_grafana.adoc (hunk 6 above)
Comment (dockerymick) [reply observability/design_grafana.adoc]: Updated

File: observability/design_grafana.adoc (hunk 8 above)
Comment (dockerymick) [reply observability/design_grafana.adoc]: Updated with **custom** to replace out-of-the-box. Thoughts? @dislbenn are there specific tasks that we want to call out?

File: observability/design_grafana.adoc (hunk 7 above)
Comment (dockerymick) [reply observability/design_grafana.adoc:233]: Great catch, Updated !

File: observability/design_grafana.adoc (hunk 8 above)
Comment (dislbenn) [reply observability/design_grafana.adoc]: @dockerymick @oafischer well the ConfigMap is ready for customers to use as is (so in a way it is OOTB), it's just that the ConfigMap will be updated after new labels are detected. Can we replace that statement with something like:

> By default, Observability ignores a subset of labels in the `observability-managed-cluster-label-allowlist` Configmap.

File: observability/design_grafana.adoc [hunk 9]
```diff
@@ -109,7 +110,7 @@ After you complete your updates for the ConfigMap, you can install it to import
 [#using-managed-cluster-labels]
 == Using managed cluster labels in Grafana
 
-When Observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, Observability ignores out-of-the-box labels in the `observability-managed-cluster-label-allowlist` ConfigMap.
+When Observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, Observability ignores custom labels in the `observability-managed-cluster-label-allowlist` ConfigMap.
```
Comment (dislbenn) [inline observability/design_grafana.adoc]: I feel like we should replace the word `custom`. I can see some people reading that as by default, any `custom` label that is added, we will ignore.

Comment (dockerymick) [reply observability/design_grafana.adoc]: Ok, what labels specifically get ignored?

Comment (dislbenn) [reply observability/design_grafana.adoc]: The labels that are ignored are the more `uncommon` labels (i.e `local-cluster`). We focused on ignoring labels that might not produce relevant results for users.

Comment (dockerymick) [reply observability/design_grafana.adoc]: Thanks for that info, i missed your [earlier comment](https://github.com/stolostron/rhacm-docs/pull/4656#discussion_r1093377489). Making the update now!

File: observability/design_grafana.adoc (hunk 8 above)
Comment (dockerymick) [reply observability/design_grafana.adoc]: Updated

**Comment by dockerymick** [comment]
Thanks @oafischer @dislbenn for the reviews and suggestions. Can you both re-review when you have an opportunity?

File: observability/design_grafana.adoc [hunk 10]
```diff
@@ -110,7 +110,7 @@ After you complete your updates for the ConfigMap, you can install it to import
 [#using-managed-cluster-labels]
 == Using managed cluster labels in Grafana
 
-When Observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, Observability ignores custom labels in the `observability-managed-cluster-label-allowlist` ConfigMap.
+When observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, bservability ignores a subset of labels in the `observability-managed-cluster-label-allowlist` ConfigMap.
```
Comment (dislbenn) [inline observability/design_grafana.adoc:113]: Observability is misspelled.

```suggestion
When observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, observability ignores a subset of labels in the `observability-managed-cluster-label-allowlist` ConfigMap.
```

File: observability/design_grafana.adoc [hunk 11]
```diff
@@ -102,6 +107,132 @@ annotations:
 +
 After you complete your updates for the ConfigMap, you can install it to import the dashboard to the Grafana instance.
 
+[#using-managed-cluster-labels]
+== Using managed cluster labels in Grafana
+
+When observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, bservability ignores a subset of labels in the `observability-managed-cluster-label-allowlist` ConfigMap.
```
Comment (dislbenn) [inline observability/design_grafana.adoc]: Observability is misspelled.

```suggestion
When observability is enabled in the hub cluster, the `observability-managed-cluster-label-allowlist` ConfigMap is created in the `open-cluster-management-observability` namespace. The ConfigMap contains a list of managed cluster labels maintained by the `observabilty-rbac-query-proxy` pod, to populate a list of label names to filter from within the _ACM - Cluster Overview_ Grafana dashboard. By default, observability ignores a subset of labels in the `observability-managed-cluster-label-allowlist` ConfigMap.
```

**Comment by dislbenn** [comment]
@dockerymick all changes look good to me, there is just one final change that needs to be made.

**Review by dislbenn — APPROVED** [review]

**Comment by dockerymick** [comment]
@dislbenn thanks for catching the misspelling ! I updated the file. Thanks for providing your approval too

File: observability/design_grafana.adoc (hunk 11 above)
Comment (dockerymick) [reply observability/design_grafana.adoc]: updated

**Review by oafischer — APPROVED** [review]
lgtm

**Comment by dockerymick** [comment]
@dislbenn @oafischer thanks for yall's approvals! Merging this PR

**Pull request merged**